plugin/defs.go

```diff
@@ -42,6 +44,30 @@ func WithSwarmService(id string) CreateOpt {
 	}
 }
 
+// WithEnv is a CreateOpt that passes the user-provided environment variables
+// to the plugin task
+func WithEnv(env []string) CreateOpt {
+	return func(p *v2.Plugin) {
+		effectiveEnv := make(map[string]string)
+		for _, penv := range p.PluginObj.Config.Env {
+			if penv.Value != nil {
+				effectiveEnv[penv.Name] = *penv.Value
+			}
+		}
+		for _, line := range env {
+			if pair := strings.SplitN(line, "=", 2); len(pair) > 1 {
```

Comment:
How come we are doing this soft validation here

Reply:
That was what was suggested by @thaJeztah